File: Algo201/problem_of_the_class/cLang/bit/bit.cpp
#include<iostream>
using namespace std;

void printBit(int arr[],int n){
	for(int i=0;i<n;i++)
		cout << arr[i] << " " ;	
	cout << endl;
}

void genBit(int n,int arr[],int i){
	if(n == i){
		printBit(arr,n);
		return ;	
	}	

	arr[i] = 0;
	genBit(n,arr,i+1);
	
	arr[i] = 1;
	genBit(n,arr,i+1);

}

int main(){
	int t;
	cin >> t;
	int arr[t];
	for(int i =0;i<t;i++)
		arr[i] = 0;

	genBit(t,arr,0);
	return 0;
}

File: Algo201/problem_of_the_class/python/bit.py
def print_bit(arr):
    for i in arr :
        print(i,end=' ')
    print()

def gen_bit(n,arr,i):
    if n == i :
        print_bit(arr)
        return 

    arr[i] = 0 
    gen_bit(n,arr,i+1)
 
    arr[i] = 1 
    gen_bit(n,arr,i+1) 

t = int(input("input:"))
arr = []

for i in range(t):
    arr.append(0)

print("output:")
gen_bit(t,arr,0)